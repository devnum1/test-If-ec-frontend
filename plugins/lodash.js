// 01EZTX9M3AWK0ZB16B6KYR56F1

import Vue from 'vue';
import _ from 'lodash';
Vue.prototype._ = _;

// To be able to use this._ or Vue._ in the components without including lodash
Vue.use(_);
