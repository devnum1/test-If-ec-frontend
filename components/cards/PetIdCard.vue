// 01EZTX9M3AWK0ZB16B6KYR56F1

<template>
  <div class="pet-id-card">
    <div class="id-card-container">
      <div class="peco-title">
        {{ $t('common.peco_title') }}
      </div>
      <div v-show="editCardPet" class="id-edit">
        <button class="id-edit__button" @click="handleClick">
          {{ $t('common.button.edit') }}
        </button>
      </div>
      <h4 class="pet-name font-weight-400" :class="editCardPet ? '' : 'mt-26'">
        <span>{{ get(petInfo, 'petName', '') }}</span>
        <span class="pet-gender">
          {{
            get(petInfo, 'gender') === constContract.pet_sex.FEMALE
              ? $t('common.chan')
              : $t('common.kun')
          }}
        </span>
      </h4>
      <div class="id-card__pet-info-container">
        <div class="pet-info-grid">
          <div class="info-row">
            <div class="card-icon">
              <i class="icon-fonts icon-fonts--cake" />
            </div>
            <div class="">
              {{ petBirthday }}
            </div>
          </div>
          <div class="info-row">
            <div class="card-icon">
              <i class="icon-fonts icon-fonts--dog-foot" />
            </div>
            <div class="">
              {{
                get(petInfo, 'petType') === constContract.pet_type.DOG
                  ? get(petInfo, 'petBreedDesc')
                  : $t('common.pet.cat')
              }}
            </div>
          </div>
          <div class="info-row">
            <div class="card-icon">
              <i class="icon-fonts icon-fonts--paper" />
            </div>
            <div class="">
              {{ compensRate }}
            </div>
          </div>
          <div class="info-row">
            <img
              class="card-logo"
              src="~/assets/images/logo/2line_bg_white.svg"
              height="36"
            />
          </div>
        </div>
        <div v-if="petImage" class="pet-image-container">
          <img class="pet-image" :src="petImage" />
        </div>
        <div v-else class="no-image-container">
          <img
            v-if="get(petInfo, 'petType') === constContract.pet_type.CAT"
            class="no-image cat-invert"
            src="~/assets/images/icon-cat.svg"
          />
          <img v-else class="no-image" src="~/assets/images/icon-dog.svg" />
        </div>
      </div>
    </div>
  </div>
</template>
<script type="text/javascript">
import contract from '~/constants/contract';
import { get } from 'lodash';
export default {
  props: {
    petInfo: {
      type: Object,
      default: () => ({}),
    },
    productInfo: {
      type: Object,
      default: () => ({}),
    },
    editCardPet: {
      type: Boolean,
      default: false,
    },
    policyNo: {
      type: String,
      default: '',
    },
    policyId: {
      type: String,
      default: '',
    },
    subscriberNo: {
      type: [String, Number],
      default: '',
    },
    noPetImage: {
      type: Boolean,
      default: false,
    },
  },
  data() {
    return {
      petImage: null,
      constContract: contract,
    };
  },
  computed: {
    petBirthday() {
      const petBirthday = get(this.petInfo, 'birthDate', '______');

      return petBirthday.substring(0, 4) + '.' + petBirthday.substring(5, 7);
    },
    compensRate() {
      const plan = get(this.productInfo, 'planCode');
      if (!plan) {
        return 'Unknown';
      }

      return (
        (plan === contract.plan_code.SEVENTY ? 70 : 50) +
        this.$t('common.compens_rate')
      );
    },
  },
  async created() {
    // DP-0019画面ではペット画像を取得しない
    if (this.noPetImage) {
      return;
    }
    const petImageKey = get(this.petInfo, 'petPhotoDoc');

    if (petImageKey) {
      const res = await this.$store.dispatch('contracts/s3download', {
        imageKey: petImageKey,
      });

      if (res.result === 1) {
        this.petImage = res.docShareUrl;
      }
    }
  },

  methods: {
    get,
    handleClick() {
      this.$store.commit('contracts/setPolicyId', this.policyId);
      this.$emit('edit');
    },
  },
};
</script>

<style lang="scss" scoped>
@import '~/assets/scss/components/cards';
</style>
