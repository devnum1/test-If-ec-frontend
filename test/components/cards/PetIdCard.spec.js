import { shallowMount, config, createLocalVue } from '@vue/test-utils';
import * as _ from 'lodash';

import PetIdCard from '@/components/cards/PetIdCard.vue';
import Vuex from 'vuex';
import VueMask from 'v-mask';

config.mocks.$t = (key) => key;
const localVue = createLocalVue();
localVue.use(Vuex);
localVue.use(VueMask);

Object.defineProperty(localVue.prototype, '_', { value: _ });

let $route = {
  hash: '',
};

const $router = {
  push: () => {},
};

let $store = {
  dispatch: () => {},
  commit: () => {},
};

const $storeReturnPromise = {
  dispatch: () => Promise.resolve(),
};

describe('PetIdCard', () => {
  const petInfo = {
    insCoverageId: 'P0002',
    insTargetProperty: {
      petName: 'ゆき',
      petType: {
        propVal: '01',
        propDispNm: '犬',
      },
      petBreed: {
        propVal: '052',
        propDispNm: '柴',
      },
      petSexCd: {
        propVal: '01',
        propDispNm: '男の子',
      },
      petWeight: 5,
      petBirthdayDiv: {
        propVal: '01',
        propDispNm: '確定',
      },
      petBirthday: '201801',
      petInsJoinFlg: {
        propVal: '01',
        propDispNm: '加入していない',
      },
    },
    document: null,
  };

  const productInfo = {
    compensRate: {
      propVal: 0.5,
      propDispNm: '50%\u88dc\u511f',
    },
  };

  let wrapper;

  const spy = jest.spyOn(PetIdCard.methods, 'handleClick');

  const factory = (computed) => {
    return shallowMount(PetIdCard, {
      propsData: {
        petInfo: petInfo,
        editCardPet: false,
        productInfo: productInfo,
      },
      mocks: {
        $store: $storeReturnPromise,
      },
      ...computed,
    });
  };

  beforeEach(() => {
    wrapper = factory();
  });

  afterEach(() => {
    wrapper.destroy();
  });

  test('is a Vue instance', () => {
    const wrapper = shallowMount(PetIdCard);
    expect(wrapper.vm).toBeTruthy();
  });

  test('match Snapshot', () => {
    expect(wrapper.html()).toMatchSnapshot();
  });

  test('Card not display edit button', () => {
    const button = wrapper.find('.id-edit__button');

    expect(button.isVisible()).toBe(false);
  });

  test('Card display edit button', () => {
    const wrapper = shallowMount(PetIdCard, {
      propsData: {
        petInfo: petInfo,
        editCardPet: true,
      },
    });
    const button = wrapper.find('.id-edit__button');

    expect(button.isVisible()).toBe(true);
  });

  test('Test handle click', async () => {
    const wrapper = shallowMount(PetIdCard, {
      propsData: {
        petInfo: petInfo,
        editCardPet: true,
      },
      localVue,
      mocks: {
        $route,
        $router,
        $store,
      },
    });
    const button = wrapper.find('.id-edit__button');
    await button.trigger('click');
    wrapper.vm.handleClick();

    expect(wrapper.emitted().edit).toBeTruthy();
  });

  test('Test pet with photo ', async () => {
    $store = {
      ...$store,
      dispatch: jest.fn().mockReturnValueOnce({ data: 'example' }),
    };

    let pet = _.cloneDeep(petInfo);
    pet.document = {
      docFileName: 'example',
    };

    const wrapper = shallowMount(PetIdCard, {
      propsData: {
        petInfo: pet,
        editCardPet: true,
      },
      localVue,
      mocks: {
        $route,
        $router,
        $store,
      },
    });
    const button = wrapper.find('.id-edit__button');
    await button.trigger('click');
    wrapper.vm.handleClick();
    expect(wrapper.vm.petImage).toBe('data:image/jpeg;base64,example');
  });

  test('Test pet with no photo ', async () => {
    $store = {
      ...$store,
      dispatch: jest.fn().mockReturnValueOnce({ data: null }),
    };

    let pet = _.cloneDeep(petInfo);
    pet.document = {
      docFileName: 'example',
    };

    const wrapper = shallowMount(PetIdCard, {
      propsData: {
        petInfo: pet,
        editCardPet: true,
      },
      localVue,
      mocks: {
        $route,
        $router,
        $store,
      },
    });
    const button = wrapper.find('.id-edit__button');
    await button.trigger('click');
    wrapper.vm.handleClick();
    expect(wrapper.vm.petImage).toBe(null);
  });
});
