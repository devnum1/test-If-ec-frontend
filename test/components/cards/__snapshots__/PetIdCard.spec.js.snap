// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PetIdCard match Snapshot 1`] = `
"<div class=\\"pet-id-card\\">
  <div class=\\"id-card-container\\">
    <div class=\\"peco-title\\">
      common.peco_title
    </div>
    <div class=\\"id-edit\\" style=\\"display: none;\\"><button class=\\"id-edit__button\\">
        common.button.edit
      </button></div>
    <h4 class=\\"pet-name font-weight-400 mt-26\\"><span>ゆき</span> <span class=\\"pet-gender\\">common.kun</span></h4>
    <div class=\\"id-card__pet-info-container\\">
      <div class=\\"pet-info-grid\\">
        <div class=\\"info-row\\">
          <div class=\\"card-icon\\"><i class=\\"icon-fonts icon-fonts--cake\\"></i></div>
          <div>
            2018.01
          </div>
        </div>
        <div class=\\"info-row\\">
          <div class=\\"card-icon\\"><i class=\\"icon-fonts icon-fonts--dog-foot\\"></i></div>
          <div>
            柴
          </div>
        </div>
        <div class=\\"info-row\\">
          <div class=\\"card-icon\\"><i class=\\"icon-fonts icon-fonts--paper\\"></i></div>
          <div>
            50common.compens_rate
          </div>
        </div>
        <div class=\\"info-row\\"><img src=\\"~/assets/images/logo/2line_bg_white.svg\\" height=\\"36\\" class=\\"card-logo\\"></div>
      </div>
      <div class=\\"no-image-container\\"><img src=\\"~/assets/images/icon-dog.svg\\" class=\\"no-image\\"></div>
    </div>
  </div>
</div>"
`;
